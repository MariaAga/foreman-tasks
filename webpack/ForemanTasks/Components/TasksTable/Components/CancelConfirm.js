import React from 'react';
import PropTypes from 'prop-types';
import { sprintf, translate as __ } from 'foremanReact/common/I18n';
import { Button } from 'patternfly-react';
import ForemanModal from 'foremanReact/components/ForemanModal';
import {
  CANCEL_CONFIRM_MODAL_ID,
  RESUME_CONFIRM_MODAL_ID,
  CANCEL_SELECTED_CONFIRM_MODAL_ID,
  RESUME_SELECTED_CONFIRM_MODAL_ID,
} from '../TasksTableConstants';

export const CancelConfirm = ({ closeModal, action, selectedRowsLen, id }) => (
  <ForemanModal title={__('Cancel Selected Tasks')} id={id}>
    <ForemanModal.Header />
    {sprintf(
      __(
        `This will cancel %s task(s), putting them in the stopped state. Are you sure?`
      ),
      selectedRowsLen
    )}
    <ForemanModal.Footer>
      <Button onClick={closeModal}>{__('No')}</Button>
      <Button
        bsStyle="primary"
        onClick={() => {
          action();
          closeModal();
        }}
      >
        {__('Yes')}
      </Button>
    </ForemanModal.Footer>
  </ForemanModal>
);

CancelConfirm.propTypes = {
  closeModal: PropTypes.func.isRequired,
  selectedRowsLen: PropTypes.number.isRequired,
  action: PropTypes.func,
  id: PropTypes.oneOf([
    CANCEL_CONFIRM_MODAL_ID,
    RESUME_CONFIRM_MODAL_ID,
    CANCEL_SELECTED_CONFIRM_MODAL_ID,
    RESUME_SELECTED_CONFIRM_MODAL_ID,
  ]).isRequired,
};

CancelConfirm.defaultProps = {
  action: () => null,
};

export default CancelConfirm;
