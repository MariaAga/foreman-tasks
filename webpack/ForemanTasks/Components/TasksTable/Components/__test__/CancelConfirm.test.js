import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { CancelConfirm } from '../CancelConfirm';
import {
  CANCEL_CONFIRM_MODAL_ID,
  CANCEL_SELECTED_CONFIRM_MODAL_ID,
} from '../../TasksTableConstants';

const baseProps = {
  closeModal: () => null,
  selectedRowsLen: 1,
  action: () => null,
};
const fixtures = {
  'renders CANCEL modal': {
    ...baseProps,
    id: CANCEL_CONFIRM_MODAL_ID,
  },
  'renders CANCEL_SELECTED modal': {
    ...baseProps,
    id: CANCEL_SELECTED_CONFIRM_MODAL_ID,
  },
};

describe('CancelConfirm', () =>
  testComponentSnapshotsWithFixtures(CancelConfirm, fixtures));
