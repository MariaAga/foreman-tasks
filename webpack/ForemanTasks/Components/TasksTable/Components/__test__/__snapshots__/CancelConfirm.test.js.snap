// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelConfirm renders CANCEL modal 1`] = `
<ForemanModal
  id="cancelConfirmModal"
  title="Cancel Selected Tasks"
>
  <Component />
  This will cancel 1 task(s), putting them in the stopped state. Are you sure?
  <Component>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      No
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      Yes
    </Button>
  </Component>
</ForemanModal>
`;

exports[`CancelConfirm renders CANCEL_SELECTED modal 1`] = `
<ForemanModal
  id="cancelSelectedConfirmModal"
  title="Cancel Selected Tasks"
>
  <Component />
  This will cancel 1 task(s), putting them in the stopped state. Are you sure?
  <Component>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      No
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      Yes
    </Button>
  </Component>
</ForemanModal>
`;
